remote-trait-object/src/transport/multiplex.rs

```diff
@@ -14,13 +14,31 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use crate::packet::{Packet, PacketView};
+use crate::packet::PacketView;
 use crate::transport::{RecvError, Terminate, TransportRecv};
 use crate::Config;
 use crossbeam::channel::{self, Receiver, Sender};
 use parking_lot::Mutex;
 use std::thread;
 
+pub struct MultiplexedRecv {
+    recv: Receiver<Result<Vec<u8>, RecvError>>,
+}
+
+impl TransportRecv for MultiplexedRecv {
+    fn recv(&self, timeout: Option<std::time::Duration>) -> Result<Vec<u8>, RecvError> {
+        if let Some(timeout) = timeout {
+            self.recv.recv_timeout(timeout).unwrap()
+        } else {
+            self.recv.recv().unwrap()
+        }
+    }
+
+    fn create_terminator(&self) -> Box<dyn Terminate> {
+        unreachable!()
```

Comment:
Using `unreachable!` is awkward. We may refactor the TransportRecv trait later.

Reply:
We can add another trait that has no `creator_terminator()` which will be used only for the server and client